pkg/operator/csi/controller/credentials/resource.go

```diff
@@ -0,0 +1,118 @@
+package credentials
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/client-go/config/clientset/versioned/scheme"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehelper"
+)
+
+const (
+	credentialsRequestGroup    = "cloudcredential.openshift.io"
+	credentialsRequestVersion  = "v1"
+	credentialsRequestResource = "credentialsrequests"
+
+	specHashAnnotation = "operator.openshift.io/spec-hash"
+)
+
+var (
+	credentialsRequestResourceGVR schema.GroupVersionResource = schema.GroupVersionResource{
+		Group:    credentialsRequestGroup,
+		Version:  credentialsRequestVersion,
+		Resource: credentialsRequestResource,
+	}
+)
+
+func readCredentialRequestsOrDie(objBytes []byte) *unstructured.Unstructured {
+	udi, _, err := scheme.Codecs.UniversalDecoder().Decode(objBytes, nil, &unstructured.Unstructured{})
+	if err != nil {
+		panic(err)
+	}
+	return udi.(*unstructured.Unstructured)
+}
+
+func applyCredentialsRequest(
```

Comment:
I'm +1 to having these in resourceapply and resourceread.  Any reason not to place them there?